lumispy/signals/luminescence_spectrum.py

```diff
@@ -562,6 +586,53 @@ def px_to_nm_grating_solver(
         "\n", "\n\t"
     )
 
+    def centroid(self, slice=None, npeaks=1):
+        """
+        Finds the centroid (center of mass) of a peak in the spectrum from the wavelength (or pixel number) and the intensity at each pixel value. It basically represents a "weighted average" of the peak.
+
+        Notes
+        -------
+        This function only works for a single peak. If you have multiple peaks, slice the signal beforehand or use the slice parameter.
+
+        TODO
+        -------
+        Implement this function for multiple peaks (npeaks = 2) by finding the top 2 peaks from mean spectrum and then returning a signal with 2 com.
+
+        Parameters
+        -------
+        slice : tuple (2)
+            An tuple representing the indices of the wavelength (start index, end index) where the peak is located. If the tuple contains int, it slices on index. Ig the tuple contains float, it slices on signal units (defualt hyperspy s.inav[:] functionality).
+
+        Returns
+        -------
+        signal : Signal1D
+            A signal containing the center of mass for each navigation pixel.
+        """
+        if slice:
+            if type(slice) != tuple:
+                raise TypeError("The `slice` parameter must be a tuple of length 2.")
+            if len(slice) != 2:
+                raise ValueError(
+                    f"The `slice` parameter must be a tuple of length 2. You passed a tuple of length {len(slice)}"
+                )
+
+            s = self.isig[slice[0] : slice[1]]
+
+        else:
+            s = self
+
+        if npeaks > 1 or npeaks < 1:
+            raise NotImplementedError("Centroid only works for one peak.")
+
+        wavelengths = s.axes_manager.signal_axes[0].axis
+        center_of_mass = s.map(com, wavelengths=wavelengths, inplace=False)
+
+        # Transfer axes metadata
+        ax = center_of_mass.axes_manager.signal_axes[0]
+        ax.units = s.axes_manager.signal_axes[0].units
+        ax.name = s.axes_manager.signal_axes[0].name
```

Comment:
Why is it necessary? The returned signal should have a signal dimension of 0, i .e. at each navigation index, there is a scalar value.

Reply:
I wanted to avoid this but you are right. I will save the units/name of the signal axes (that disappears) in the title of the BaseSignal that stays.